head/src/main/kotlin/io/qalipsis/core/head/report/thymeleaf/ThymeleafReportServiceImpl.kt

```diff
@@ -127,24 +147,41 @@ internal class ThymeleafReportServiceImpl(
             setVariable("chartImages", chartImages)
             // Configure the path to read the font family.
             setVariable("fontUrl", fontUrl)
+            // Configure the path for the time svg icon.
+            setVariable("timeImage", assets["timeImage"])
+            // Configure the path for the user svg icon.
+            setVariable("userImage", assets["userImage"])
+            // Configure the path for the css styles.
+            setVariable("stylePath", assets["stylePath"])
         }
+    }
 
     /**
      * Generates a pdf template from HTML string.
      *
      * @param html HTML template to be rendered
      */
+    @KTestable
     private suspend fun generatePdfFromHtml(html: String): ByteArray {
         try {
             require(html.isNotEmpty()) { "HTML File is empty" }
+            val document = Jsoup.parse(html, CHARACTER_ENCODING)
+            document.outputSettings().syntax(Document.OutputSettings.Syntax.xml)
             val outputStream = ByteArrayOutputStream()
-            ITextRenderer().apply {
-                setDocumentFromString(html)
-                layout()
-                createPDF(outputStream)
+            outputStream.use {
+                ITextRenderer().apply {
+                    val sharedContext = sharedContext
+                    sharedContext.isPrint = true
+                    sharedContext.isInteractive = false
+                    // @FIXME This should help with the image rendering but it throws an exception. Uncomment to debug
+                    sharedContext.replacedElementFactory =
+                        SVGReplacedElementFactory(sharedContext.replacedElementFactory)
+                    setDocumentFromString(document.html())
+                    layout()
+                    createPDF(outputStream)
```

Comment:
When you pass the whole stream, no need to do it in a use{} block. use{} is only when you read and process the stream internally, not just passing it to a further function that will read it.

Reply:
The `createPDF` function is only available within the `ITextRenderer().apply {}` so it has to be within that block. nless I create an instance of ITextRenderer() outside the use and call the createPDF there. i.e 
```
val itextRenderer = ITextRenderer()
val outputStream = ByteArrayOutputStream()
            outputStream.use {
                ITextRenderer().apply {
                layout()
                }
            }
 itextRenderer.createPDF(outputStream)
```
This is how its going to be.